### PR TITLE
AppFormer application provisioning

### PR DESCRIPTION
**Pull request by wmedvede**
- improves the appformer app provisioning by:
    - adding a wizard
    - enables the selection of datasources in the target server
    - automatically configures the local server

### CODE REVIEW

**Comment by wmedvede** [comment]
Hello @pefernan would you mind review this PR?
Thanks.

File: appformer-showcase/appformer-webapp/pom.xml [hunk 1]
```diff
@@ -1520,7 +1520,7 @@
 
               <!-- AppFormer -->
               <compileSourcesArtifact>org.kie.appformer:appformer-form-modeler-rendering-api</compileSourcesArtifact>
-              <compileSourcesArtifact>org.kie.appformer:appformer-form-modeler-dynamic-renderer-client</compileSourcesArtifact>
+              <!--<compileSourcesArtifact>org.kie.appformer:appformer-form-modeler-dynamic-renderer-client</compileSourcesArtifact>-->
```
Comment (pefernan) [inline appformer-showcase/appformer-webapp/pom.xml]: I think we can remove this line instead of comment it.

**Comment by wmedvede** [comment]
@pefernan fixed, thanks.

**Comment by wmedvede** [comment]
jenkins please retest this

**Comment by csadilek** [comment]
@wmedvede @pefernan is it possible to add some test coverage for this? The failing tests here don't look related on first sight?

**Comment by wmedvede** [comment]
Hi @csadilek @pefernan  I believe that the failing tests are not directly related to this PR. Seems like they might be some collateral effect of the migration of the Roaster library. Just a feeling, because I see that all the failing stuff is related to some Roaster based generation in appformer. I believe we can merge this one, and we can then try to reproduce the error locally.

**Comment by pefernan** [comment]
@wmedvede @csadilek This issues are defnitelly related to the Roaster version update. It seems that they changed the implementation of some methods that now are working slightly different than the one we had before. I have a fix for this on my fork, but I'll need some more time to test that the form generation stuff is working well. Gimme one more day to check that all keeps working fine and I'll send the PR.

**Comment by wmedvede** [comment]
Sorry @pefernan , it was for sure my fault. But I don't understand why it fails now at this point and not before.

**Comment by pefernan** [comment]
@wmedvede @csadilek These PR's are fixing the issues on source generation engine.

https://github.com/droolsjbpm/kie-wb-common/pull/677
https://github.com/droolsjbpm/appformer/pull/72

**Comment by csadilek** [comment]
jenkins retest this.

**Pull request merged**